src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion.py

```diff
@@ -116,6 +119,7 @@ def __init__(
         scheduler: KarrasDiffusionSchedulers,
         safety_checker: StableDiffusionSafetyChecker,
         feature_extractor: CLIPImageProcessor,
+        image_encoder: CLIPVisionModelWithProjection = None,
```

Comment:
@patrickvonplaten defaulting to None breaks the serialization compatibility (as demonstrated in https://github.com/huggingface/diffusers/pull/4944#issuecomment-1802081374). Not defaulting to None breaks the existing tests. Any suggestions?

Reply:
I think we should try to update the tests instead -

Reply:
Same.